contracts/interfaces/ICover.sol

```diff
@@ -9,27 +9,44 @@ interface ICover is IERC721 {
   /* ========== DATA STRUCTURES ========== */
 
   struct StakingPool {
-    uint160 id;
-    uint96 bookedAmount;
+    address poolAddress;
```

Comment:
Optimize using `poolId` and `CREATE2`. We'll need to work on the minimal proxy contract (what I have in mind is not quite standard).